internal/monitor/metadata.go

```diff
@@ -0,0 +1,104 @@
+package monitor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+)
+
+type Metadata struct {
+	ClusterID string `json:"clusterId"`
+	LastStart int64  `json:"lastStart"`
+}
+
+func (m *Metadata) Save(file string) error {
+	if file == "" {
+		// if monitor is running standalone or with an old chart version, and saving of
+		// metadata is not configured, we don't need to do anything here
+		return nil
+	}
+	contents, err := json.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("marshaling: %w", err)
+	}
+	return os.WriteFile(file, contents, 0o600)
+}
+
+func (m *Metadata) Load(file string) error {
+	contents, err := os.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("reading file: %w", err)
+	}
+
+	if err := json.Unmarshal(contents, m); err != nil {
+		return fmt.Errorf("file: %v content: %v parsing json: %w", file, string(contents), err)
+	}
+	return nil
+}
+
+// watchForMetadataChanges starts a watch on a local file for updates and returns changes to metadata channel. watcher stops when context is done
+func watchForMetadataChanges(ctx context.Context, log logrus.FieldLogger, metadataFilePath string) (chan Metadata, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("setting up new watcher: %w", err)
+	}
+	updates := make(chan Metadata, 10)
```

Comment:
Why it is a buffering channel?

Reply:
added non blocking write to channel and tests